src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/LinkedList.cs

```diff
@@ -0,0 +1,192 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Diagnostics
+{
+    internal sealed partial class LinkedListNode<T>
+    {
+        public LinkedListNode(T value) => Value = value;
+        public T Value;
+        public LinkedListNode<T>? Next;
+    }
+
+    // We are not using the public LinkedList<T> because we need to ensure thread safety operation on the list.
+    internal sealed class LinkedList<T> : IEnumerable<T>
```

Comment:
We should name it something other than LinkedList, even though it's in a different namespace.

Reply:
to note, I didn't introduce this in my changes, I just moved it to its own file. but I agreeing renaming is good.